### PR TITLE
feat: integration test for SignIn flow

### PR DESCRIPTION
**Pull request by nicolantean**
# :star: Feature

---

#### :pencil2: Description:

Added integration test for Sign In Screen, including success and failure signing events.

---

#### iOS

https://github.com/xmartlabs/flutter-template/assets/49296136/833e282e-7c0e-43d2-a510-cd3215b36600

### CODE REVIEW

File: integration_test/signin_test.dart [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  late MockSessionRepository mockSessionRepository;
+  late AppRouter appRouter;
+  late BuildContext context;
+  late StreamController<AuthenticationStatus> statusController;
+
+  setUp(() async {
+    await app.initSdks();
```
Comment (fedecor9) [inline integration_test/signin_test.dart:28]: What do you think about having this setup with the unregister as a helper?

**Review by fedecor9 — APPROVED** [review]
Nice

File: integration_test/test/signin_test.dart [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import 'package:flutter_template/ui/section/error_handler/global_event_handler_cubit.dart';
+import 'package:flutter_template/ui/signin/signin_screen.dart';
+import 'package:flutter_template/ui/welcome/welcome_screen.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../common/general_helpers.dart';
+import '../common/repository_mocks.dart';
```
Comment (f7deleon) [inline integration_test/test/signin_test.dart:17]: check this out

File: integration_test/test/signin_test.dart [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        .thenAnswer((_) => statusController.stream);
+    appRouter = AppRouter(mockSessionRepository);
+    await commonSetup(mockSessionRepository, appRouter);
+  });
+
+  tearDown(() {
+    DiProvider.instance.reset();
+    statusController.close();
+  });
+  group('SignIn screen tests', () {
```
Comment (f7deleon) [inline integration_test/test/signin_test.dart:40]: can we extract this into a method?

Comment (f7deleon) [reply integration_test/test/signin_test.dart:40]: I think this case isn't long enough but if you are testing a form you might end up with a very long method main that test everything

**Review by f7deleon — APPROVED** [review]

File: integration_test/test/signin_test.dart (hunk 2 above)
Comment (nicolantean) [reply integration_test/test/signin_test.dart:17]: These files are in the test folder and outside of lib, so the unique way to import them is using a relative path.

File: integration_test/test/signin_test.dart (hunk 3 above)
Comment (nicolantean) [reply integration_test/test/signin_test.dart:40]: Do you mean to divide each group into different methods?

Comment (f7deleon) [reply integration_test/test/signin_test.dart:40]: yes

File: integration_test/test/signin_test.dart [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    testWidgets('SignIn with wrong credentials', (WidgetTester tester) async {
+      mockSignInUser(false, mockSessionRepository, statusController);
+      await Future.delayed(const Duration(seconds: 2));
```
Comment (mirland) [inline integration_test/test/signin_test.dart:43]: whY?

File: flavorizr.yaml [hunk 5]
```diff
@@ -1,3 +1,7 @@
+// IMPORTANT
+// To keep integration test working, after excecuting flavorizr for creating more schemas
+// you need to remove FLUTTER_TARGET line from all .xcconfig files on ios/Flutter directory
+
```
Comment (mirland) [inline flavorizr.yaml:4]: comments are with #

File: integration_test/test/signin_test.dart [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        find.widgetWithText(TextField, context.localizations.password),
+        password,
+      );
+      await tester.tap(
+        find.widgetWithText(TextButton, context.localizations.sign_in),
+      );
+      await tester.pumpAndSettle();
+      expect(find.byType(AlertDialog), findsOneWidget);
+    });
+    testWidgets('SignIn with correct credentials', (WidgetTester tester) async {
```
Comment (mirland) [inline integration_test/test/signin_test.dart:63]: new line

File: scripts/integration_test.sh [hunk 7]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+flutter drive --driver=test_driver/integration_test.dart --target=integration_test/test/signin_test.dart --flavor dev
```
Comment (mirland) [inline scripts/integration_test.sh:3]:
```suggestion
flutter drive --driver=test_driver/integration_test.dart --target=integration_test/test/signin_test.dart --flavor dev

```

File: ios/Runner.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -1333,6 +1344,7 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
 				CURRENT_PROJECT_VERSION = "$(FLUTTER_BUILD_NUMBER)";
+				DEVELOPMENT_TEAM = 6F2G55XL63;
```
Comment (mirland) [inline ios/Runner.xcodeproj/project.pbxproj:1347]: rn (in the file)

File: .vscode/launch.json [hunk 9]
```diff
@@ -33,6 +25,13 @@
         "--web-port=3003",
         "--dart-define=ENV=staging"
       ]
+    },
+    {
+      "name": "Integration Test",
+      "program": "integration_test/test",
+      "request": "launch",
+      "type": "dart",
+      "args": ["--flavor=dev"]
```
Comment (mirland) [inline .vscode/launch.json:34]: web port? not needed?

**Review by mirland — APPROVED** [review]

File: .vscode/launch.json (hunk 9 above)
Comment (nicolantean) [reply .vscode/launch.json:34]: It is not needed

**Review by f7deleon — APPROVED** [review]

**Pull request merged**